### ISSUE TITLE
CF DEPLOYMENT NEVER WORKS

### ISSUE THREAD

**Issue by I501307** [SAP-samples/cap-sflight#360]
app gets deployed to cf but not at all in working conditon : {"statusCode":401,"code":"401","message":"Unauthorized"}
can someone please share why am I getting this error?

**Comment by gregorwolf** [comment]
Please provide more details where exactly you get this error. Are you trying the NodeJS or Java variant?

**Comment by srinikitha09** [comment]
> Please provide more details where exactly you get this error. Are you trying the NodeJS or Java variant?

We used mbt tool for the deployment. When we try to click on the hyperlinks (Books, Authors etc.,) as shown in the image, it throws unauthorized error.
![image](https://user-images.githubusercontent.com/26958230/205279366-d96be92a-77cb-4469-8ac7-c122b076f513.png)

**Comment by gregorwolf** [comment]
Have you [assigned the Role Collection](https://github.com/SAP-samples/cap-sflight#assign-role-collection)?

**Comment by srinikitha09** [comment]
> Have you [assigned the Role Collection](https://github.com/SAP-samples/cap-sflight#assign-role-collection)?

Thank you for the hint. Unfortunately, we do not have enough permissions on the level of subaccount to assign those role collections. We will try to deploy to another account and try in that way and will get back to you. Thank you :).

**Comment by I501307** [comment]
This still doesn't work : 
{"statusCode":401,"code":"401","message":"Unauthorized"}

**Comment by gregorwolf** [comment]
Do you see any further error messages in the approuter / srv logs?

**Comment by srinikitha09** [comment]
> Unfortunately you haven't provided detailed information what endpoint you're accessing. Seems that you try to access the srv endpoint directly. That does not work unless your browser would send a valit JWT token in the [Authorization](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization) header as a Bearer token. You have to access the app via the HTML5 application that gets deployed to the managed approuter. I just did deploy this app to my BTP trial account and it works like a charm.

Yes, you are right! We are trying to access the srv endpoint directly. But, when try to access it via HTML5 application it is forcing us to subscribe to either of these services. is it really needed to subscribe to these services to run or see something through HTML5 applications?
![image](https://user-images.githubusercontent.com/26958230/205854223-a846873d-4652-4a82-86aa-6eb52b3c8aba.png)

**Comment by gregorwolf** [comment]
As this project doesn't provide a standalone approuter you have to subscribe to one of the mentioned services. The cheapest or even free one is the Launchpad Service. You might create a new issue in this project and suggest the addition of a standalone approuter.

**Comment by I501307** [comment]
> Unfortunately you haven't provided detailed information what endpoint you're accessing. Seems that you try to access the srv endpoint directly. That does not work unless your browser would send a valit JWT token in the [Authorization](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization) header as a Bearer token. You have to access the app via the HTML5 application that gets deployed to the managed approuter. I just did deploy this app to my BTP trial account and it works like a charm.

Do we have any information or documentation on how to deploy to cf using the steps that you carried out ? Is there any project that works out of the box?

**Comment by gregorwolf** [comment]
Maybe we can also close this issue @I501307 ?
